File: app/src/main/java/com/czy/androidutils/sytem/utils/DisplayUtils.java
package com.czy.androidutils.sytem.utils;

import android.content.Context;

/**
 * 作者：叶应是叶
 * 时间：2018/3/11 16:30
 * 描述：
 */
public class DisplayUtils {

    /**
     * dp值转换为px值
     */
    public static int dp2px(Context context, float dpValue) {
        return (int) (dpValue * context.getResources().getDisplayMetrics().density + 0.5f);
    }

    /**
     * px值转换为dp值
     */
    public static int px2dp(Context context, float pxValue) {
        return (int) (pxValue / context.getResources().getDisplayMetrics().density + 0.5f);
    }

    /**
     * px值转换为sp值
     */
    public static int px2sp(Context context, float pxValue) {
        return (int) (pxValue / context.getResources().getDisplayMetrics().scaledDensity + 0.5f);
    }

    /**
     * sp值转换为px值
     */
    public static int sp2px(Context context, float spValue) {
        return (int) (spValue * context.getResources().getDisplayMetrics().scaledDensity + 0.5f);
    }

}

File: basis/src/main/java/leavesc/hello/basis/utils/DisplayUtils.kt
package leavesc.hello.basis.utils

import android.content.Context

/**
 * 作者：leavesC
 * 时间：2019/6/21 10:46
 * 描述：
 */
object DisplayUtils {

    /**
     * dp值转换为px值
     */
    fun dp2px(context: Context, dpValue: Float): Int {
        return (dpValue * context.resources.displayMetrics.density + 0.5f).toInt()
    }

    /**
     * px值转换为dp值
     */
    fun px2dp(context: Context, pxValue: Float): Int {
        return (pxValue / context.resources.displayMetrics.density + 0.5f).toInt()
    }

    /**
     * px值转换为sp值
     */
    fun px2sp(context: Context, pxValue: Float): Int {
        return (pxValue / context.resources.displayMetrics.scaledDensity + 0.5f).toInt()
    }

    /**
     * sp值转换为px值
     */
    fun sp2px(context: Context, spValue: Float): Int {
        return (spValue * context.resources.displayMetrics.scaledDensity + 0.5f).toInt()
    }

}
